src/docfx/build/Build.cs

```diff
@@ -80,8 +79,20 @@ Docset GetBuildDocset()
                 var sourceDependencies = new ConcurrentDictionary<Document, List<DependencyItem>>();
                 var bookmarkValidator = new BookmarkValidator();
                 var manifestBuilder = new ManifestBuilder();
+                var monikersMap = new ConcurrentDictionary<Document, List<string>>();
 
-                await ParallelUtility.ForEach(docset.BuildScope, BuildOneFile, ShouldBuildFile, Progress.Update);
+                await ParallelUtility.ForEach(
+                    docset.BuildScope.Where(doc => doc.ContentType != ContentType.TableOfContents),
+                    BuildOneFile,
+                    ShouldBuildFile,
+                    Progress.Update);
+
+                // Build TOC: since toc file depends on the build result of every node
+                await ParallelUtility.ForEach(
```

Comment:
>ParallelUtility [](start = 22, length = 15)

add MonikersMap to BuildFile Output and BuildToc Input, then we can clear see the dependency between these two step. Currently I can't see and dependency between this two step?

Reply:
right, maybe extract two methods where `BuildNoneTocFiles` returns a `MonikerMap` and `BuildTocFiles` takes a `MonikerMap` as an input.